src/almanack/metrics/data.py

```diff
@@ -502,3 +503,27 @@ def process_repo_for_analysis(
 
     finally:
         shutil.rmtree(temp_dir)
+
+
+def _get_almanack_version() -> str:
+    """
+    Seeks the current version of almanack using either pkg_resources
+    or dunamai to determine the current version being used.
+
+    Returns:
+        str
+            A string representing the version of almanack currently being used.
+    """
+
+    try:
+        # attempt to gather the development version from dunamai
+        # for scenarios where almanack from source is used.
+        import dunamai
+
+        return dunamai.Version.from_any_vcs().serialize()
+    except (RuntimeError, ModuleNotFoundError):
+        # else grab a static version from __init__.py
+        # for scenarios where the built/packaged almanack is used.
+        import almanack
+
+        return almanack.__version__
```

Comment:
Just FYI, it seems that if you install the almanack as an editable package, whatever's occurring during build to rewrite the version number doesn't happen, and you get `0.0.0`, which I assume comes from the `pyproject.toml`'s version field.

I think what I'm describing is an unusual situation and whoever runs it like that would probably be prepared to see an incongruent version number, but I just wanted to point it out in case it mattered.

Reply:
Thanks and good point! I hadn't thought about that use case. I'd like to explore this further in an iterative fashion, improving this if we can. The gist is that `dunamai` relies upon git tags, and `poetry-dynamic-versioning` relies on the package build process in order to inject the version number, so we haven't covered all installation scenarios with the editable package scenario.

Reply:
Created #185 to help address this in the future. Resolving for now.